ext/spl/spl_observer.c

```diff
@@ -507,7 +519,16 @@ PHP_METHOD(SplObjectStorage, count)
 	}
 
 	if (mode == COUNT_RECURSIVE) {
-		RETURN_LONG(php_count_recursive(&intern->storage));
+		zend_long ret;
+
+		if (mode != COUNT_RECURSIVE) {
+			ret = zend_hash_num_elements(&intern->storage);
+		} else {
+			ret = php_count_recursive(&intern->storage);
+		}
+
+		RETURN_LONG(ret);
+		return;
```

Comment:
Spurious change?

Reply:
I removed this dead code in an earlier commit to 8.1 and somehow this got restored on this branch. Will fix